### DISCUSSION TITLE
Sync 3 turning on and shuting down right after

### DISCUSSION THREAD

**Discussion by maticlms** [EtoTen/stm32_can_sim#2] [Q&A]
Hi there! 

Did anyone try any other codes for booting the sync 3 module. Mine worked at first but then after some time it started turning on and shutting off and giving battery saving countdowns....its hard to work with the sync like that...

Did anyone have the same issue and fixed it?

**Comment by EtoTen** [comment]
Strange, maybe you have a loose wire or something?
I would expect it to either fully work or not work, but not what you describe.
You could try changing the codes that are used around... commenting / un-commenting them from the list in "can_messages"

**Reply by maticlms** [reply]
I did try to make changes to the code. And before I carry on, 100 times thx. for even making the code and the schematic for wiring. When the splash screen shows and sync boots up I can see the home screen for mby. 20 seconds, but then it just turns off like it lost ignition, I can see the splash screen of a turning-off car again and it keeps turning ON/OFF so booting up, loading, and shuts down...or sometimes I get that annoying power saving countdown and I can't do anything about it...

**Reply by EtoTen** [reply]
No problem, glad to help people out.

That is really weird though. What specific sync 3 version do you have? 

try this:
``` cpp
static can_messages cMessages[2] = { 
 // {{0x048},{0x00, 0x00, 0x00, 0x00, 0x07, 0x00, 0xE0, 0x00}},  // turn on C1MCA - this turned most units on
 // {{0x3B3},{0x00, 0x00, 0x00, 0x00, 0x07, 0x00, 0xE0, 0x00}},  // turn on CGEA 1.3 - newer protocol, this actually made some APIMs turn off so it is commented out
//  {{0x044},{0x88, 0xC0, 0x0C, 0x10, 0x04, 0x00, 0x02, 0x00}},  //sync4 turn on

  {{0x3B3},{0x41, 0x00, 0x00, 0x00, 0x4c, 0x00, 0xE0, 0x00}},  // turn CGEA 1.3 - second option, seems to work better
  {{0x109},{0x00, 0x03, 0x01, 0x00, 0x00, 0x00, 0x00, 0x28}}   // Accessory ON, Gear Park, Speed 0 - this needed or else the APIM will default to driving mode and will lock out many settings
 };
```
then this: 
```cpp
static can_messages cMessages[2] = { 
 {{0x048},{0x00, 0x00, 0x00, 0x00, 0x07, 0x00, 0xE0, 0x00}},  // turn on C1MCA - this turned most units on
 // {{0x3B3},{0x00, 0x00, 0x00, 0x00, 0x07, 0x00, 0xE0, 0x00}},  // turn on CGEA 1.3 - newer protocol, this actually made some APIMs turn off so it is commented out
//  {{0x044},{0x88, 0xC0, 0x0C, 0x10, 0x04, 0x00, 0x02, 0x00}},  //sync4 turn on
//  {{0x3B3},{0x41, 0x00, 0x00, 0x00, 0x4c, 0x00, 0xE0, 0x00}},  // turn CGEA 1.3 - second option, seems to work better
  {{0x109},{0x00, 0x03, 0x01, 0x00, 0x00, 0x00, 0x00, 0x28}}   // Accessory ON, Gear Park, Speed 0 - this needed or else the APIM will default to driving mode and will lock out many settings
 };
```
then this: 
```cpp
static can_messages cMessages[2] = { 
// {{0x048},{0x00, 0x00, 0x00, 0x00, 0x07, 0x00, 0xE0, 0x00}},  // turn on C1MCA - this turned most units on
 {{0x3B3},{0x00, 0x00, 0x00, 0x00, 0x07, 0x00, 0xE0, 0x00}},  // turn on CGEA 1.3 - newer protocol, this actually made some APIMs turn off so it is commented out
//  {{0x044},{0x88, 0xC0, 0x0C, 0x10, 0x04, 0x00, 0x02, 0x00}},  //sync4 turn on
//  {{0x3B3},{0x41, 0x00, 0x00, 0x00, 0x4c, 0x00, 0xE0, 0x00}},  // turn CGEA 1.3 - second option, seems to work better
  {{0x109},{0x00, 0x03, 0x01, 0x00, 0x00, 0x00, 0x00, 0x28}}   // Accessory ON, Gear Park, Speed 0 - this needed or else the APIM will default to driving mode and will lock out many settings
 };
```

**Reply by maticlms** [reply]
Thank you for all your help! I have tried the combinations like you did here, I eaven tried to change parked mode, neutral, drive, eaven tried to simulate a moving care, and put in code for speed values, but it did not help... Sync 3 version was 3.0.19205 and here is the link the photo of the lable on my APIM: https://mega.nz/file/tipCHLrI#Zcfw2KVvAU4gLm6xFqNat07sCautXa_o0RSq2C85GFg. Alwso, while i got it working for a bit longer I tried to update over USB with CyanLabs to the newest 3.4 version and bricked it. Do you have any experiance with eMMC programing? Alwso did you try programing on the bench with Forscan I forgot to trie it before I bricked it...but now I cant conect to the module anymore because it is in a boot loop... Does forscan not mind that there is no actual car or that there are modules missing? Link of the boot loop video: https://mega.nz/file/BrBm3YbB#l4p-L9MJoVMDo-oWNLbP1XicgJf91dHpCDbSzehXE9o

**Reply by EtoTen** [reply]
mega link is: "The file you are trying to download is no longer available" 

Yes I have done the eMMC flashing, its pretty simple if you can solder.... there are diagrams of which test points to hook jumper wires to. You also need an EMMC to SD card break out card. I used this one https://www.exploitee.rs/index.php/Exploitee.rs_Low_Voltage_e-MMC_Adapter You also need to find a full image of someone's flash. This is a little more challenging. I have seen some people post them on drive.ru. 
This is worth checking out also: https://community.cyanlabs.net/t/apim-firmware-flash-rescue/4691

Per forscan, it sort of minds. If you had a profile previously saved with the module in the car, it will let you continue with only the APIM. However with no profile it will not build one w/o the other modules being present. 

I feel like per your original issue there is something wrong with the wiring. Double check everything.

**Reply by maticlms** [reply]
Oh really? Just before I carry on...Do you mby have Discord, it would be easier to communicate further... So...the wiring was not the issue of my problems I'm 80% sure...but, unfortunately, I created a profile for my module in the car, but then deleted all my profiles, idk why. Really you used that breakout SD board...but doesn't it have not enough pins??? Because I have seen ppl connect sync3 emmc directly to the contacts of a USB to an sd card reader...(like this https://media.ffclub.ru/th/up370967-connect.jpg). What program did you use to upload the image, there are many ways to do a partition copy 1for1. Do you by any chance have the file for a non-navi APIM, you would help me out greatly. The flash rescue is a no go for my instance because I do not have a car to do it on...and I think my module is too far gone, because now when sending a boot signal with STM32 a small clicking sound can be heard in the module or screen, not really sure, but I think that is a boot loop... Also, some guy said to me that after flashing he uses IDS to install some kind of factory files almost like EEPROM onto the module...do you have any idea about that? Thank you greatly again for all the info. I'm an electrical engineer to be, and electronics Is my passion, but they don't teach you embedded systems in school, as a matter of fact, they don't teach you shi***.

**Reply by EtoTen** [reply]
Maybe I used an RT809h instead, but I am pretty sure you only need data0 for basic r/w and the rest just allow for faster access. You can use "dd"  in Linux to push a raw, complete image.

Check/switch which pin you use for ground. Also, are you using a lab power supply? What is the amperage draw when its doing that looping stuff? Is your PSU stable?

Check this method too: https://community.cyanlabs.net/uploads/short-url/1tbE7DL4uQPHEHEJLvYMb6WX9cx.pdf

The APIM consists of 2 boards...
**Board 1**, the motherboard interfaces with the CAN Bus, provides power to the second board, and controls the screen. It has it's own personalization in the form of firmware files that can be flashed via IDS or ForScan. 
**Board 2** is the "compute module" , this board has the flash and CPU used for the heavy lifting. The Boards can have 8gb, 16gb, 32gb, or 64gb of flash. The amount of flash determines if they can support built in NAV or not. The image on these can also be flashed via IDS or ForScan.

I do not have any full images saved, sorry. I  haven't really used discord in a while.

Here are some more diagrams I have saved:

![up370967-500026800593_280769](https://user-images.githubusercontent.com/22832694/215182739-fbca0905-66a1-433e-8de4-a89f69799809.jpg)
![C7409211-7DDA-458F-9595-19F99E0486CC](https://user-images.githubusercontent.com/22832694/215183184-5d113ae3-28c5-46dd-b8e6-2f0fc1ed59b4.jpeg)
![B8695E60-C6F6-4D83-9750-813E790C3512](https://user-images.githubusercontent.com/22832694/215183239-62ff3bf8-d8ac-4ea2-a8c6-b9df954c9fe0.jpeg)
![up370967-400084400329_53017](https://user-images.githubusercontent.com/22832694/215183462-ae29f6fa-fa59-4587-a66f-0077afa56e91.jpg)

**Reply by maticlms** [reply]
Yes I have a stable bench power supply 50V/5A. During the boot loop i think it is around 0.4A but the peek is probobly higher, but I didnt conect the PSU to pc to look at he current graph... Thank you so much for the photos...

**Reply by EtoTen** [reply]
> Yes I have a stable bench power supply 50V/5A. During the boot loop i think it is around 0.4A but the peek is probobly higher, but I didnt conect the PSU to pc to look at he current graph... Thank you so much for the photos...

50vdc???? It's supposed to be +12vdc

**Reply by maticlms** [reply]
It's an industrial bench power supply, it can be any voltage and current I chose...It has PC software so I can monitor real-time current consumption. Don't worry Im not insane to put 250W into a Sync module.
